book/src/qobject/qobject_struct.md

```diff
@@ -5,42 +5,134 @@ SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
 SPDX-License-Identifier: MIT OR Apache-2.0
 -->
 
-# QObject Marked Struct
+# `#[cxx_qt::qobject]` Macro - Defining QObjects in Rust
+
+Defining QObjects is at the heart of CXX-Qt.
+Therefore `#[cxx_qt::qobject]` can be considered the most important macro in CXX-Qt.
+
+## Requirements
+- Like most other CXX-Qt macros, it can only be used from within a [`#[cxx_qt::bridge]`](./bridge-macro.md).
+- The `#[cxx_qt::qobject]` macro must be placed on a Rust struct.
+- The struct must `impl Default`, so that it can be constructed as part of a QObject.
+
+## Effects
+Adding the macro to a Rust struct `MyObject` has a few effects.
+
+However, first it's important to mention that this macro **does not modify the contents of the struct in any way!**
+This means you can always rely on the struct to behave like any other Rust struct.
+You can use it in normal Rust code, without interacting with Qt in any way.
+
+The macro does multiple other things for you though:
+- Generate a C++ QObject subclass that wraps the `MyObject` Rust struct.
+- Expose the generated QObject subclass as `qobject::MyObject`. (For details see the [`qobject::T`](./cpp_object.md) page).
+- Generate getters/setters for all fields.
+- Generate `Q_PROPERTY`s for all fields that are marked as `#[qproperty]`.
+- Generate signals if paired with a [`#[cxx_qt::qsignals]` enum](./signals_enum.md).
+
+### `base` attribute
+Use the `base` attribute to specify a C++ class that the C++ QObject will inherit from.
+The base class must be a QObject!
```

Comment:
```suggestion
The base class must inherit from QObject.
```

Reply:
I considered this phrasing initially as well.
The thing I'm unsure about is whether people will understand that "inherit" is transitively here.
i.e. your base class doesn't have to directly inherit from `QObject`, it just needs to at some point inherit from QObject.

Reply:
How about:

The base class must inherit from QObject (this can be done indirectly from a base class that inherits QObject).